src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs

```diff
@@ -12,6 +12,7 @@ public class KnowledgeComponentMastery : EventSourcedAggregateRoot
         public int LearnerId { get; private set; }
         public bool IsPassed { get; private set; }
         public bool IsCompleted { get; private set; }
+        public bool IsSatisfied { get; private set; }
```

Comment:
Satisfied is from cmi5 terminology, it basically means that the 'moveOn' criteria for the KC has been met. In our case, currently the 'moveOn' criteria for all KCs is 'Passed' (but I believe we discussed switching it to 'Completed' for the test with students).
Unlocked just means that the Learner can access it (so, all of it's children are Satisfied). Although I do suppose that doesn't really need to be an explicit state since it can be calculated at any time.